src/libraries/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -1222,14 +1243,15 @@ private void SetRequestHandleProxyOptions(WinHttpRequestState state)
                     {
                         Debug.Assert(state.WindowsProxyUsePolicy == WindowsProxyUsePolicy.UseCustomProxy);
                         updateProxySettings = true;
-                        if (state.Proxy.IsBypassed(uri))
+
+                        Uri? proxyUri = state.Proxy.IsBypassed(uri) ? null : state.Proxy.GetProxy(uri);
+                        if (proxyUri == null)
```

Comment:
This one is effectively a bug fix. I think the case is simple enough here.

Reply:
This would be a bug in the GetProxy implementation, but it's fine to be defensive.